### ISSUE TITLE
FirebaseError: Installations: Missing App configuration value: "apiKey" (installations/missing-app-config-values).

### ISSUE THREAD

**Issue by abhishekpatel946** [abhishekpatel946/Smart-Vehicle-Fleet-Manager#56]
**Describe the bug**

**To Reproduce**
1. Go to 'https://smart-vehicle-fleet-manager.web.app/'
2. See an error in the console

**Screenshots**
```
It looks like you're using the development build of the Firebase JS SDK.
When deploying Firebase apps to production, it is advisable to only import
the individual SDK components you intend to use.

For the module builds, these are available in the following manner
(replace <PACKAGE> with the name of a component - i.e. auth, database, etc):

CommonJS Modules:
const firebase = require('firebase/app');
require('firebase/<PACKAGE>');

ES Modules:
import firebase from 'firebase/app';
import 'firebase/<PACKAGE>';

Typescript:
import * as firebase from 'firebase/app';
import 'firebase/<PACKAGE>';
```

```
Uncaught FirebaseError: Installations: Missing App configuration value: "apiKey" (installations/missing-app-config-values).
    at J (extract-app-config.ts:54)
    at extract-app-config.ts:41
    at e.instanceFactory (index.ts:48)
    at e.getOrInitializeService (provider.ts:194)
    at e.getImmediate (provider.ts:95)
    at e.instanceFactory (index.ts:61)
    at e.getOrInitializeService (provider.ts:194)
    at e.getImmediate (provider.ts:95)
    at e._getService (firebaseApp.ts:127)
    at e.<computed> [as performance] (firebaseNamespaceCore.ts:228)
```

**Desktop (please complete the following information):**
 - OS: [Linux]
 - Browser [Chrome]
 - Version [92.0.4515.107 (Official Build) (64-bit)]

**Additional context**

**Comment by atom19-i** [comment]
@abhishekpatel946 i can help you fix this issue. i would need a screenshot and you can send it as follows: open firebase console(of your project) -> go to project settings -> scroll down to see (your project name) under Web apps sections -> you will see **const firebase config** _screenshot that whole function_ and send it here or to my mail.

**Comment by abhishekpatel946** [comment]
@atom19-i Yeah that's the firebase config section contains api keys and all. Are you sure is it really needed if it's need to check it out our codebase their is available.

**Comment by atom19-i** [comment]
@abhishekpatel946 that'll help,yes. I have worked on similar error in a project. assign me this issue ; i will reproduce it and will do my best to solve it.

**Comment by atom19-i** [comment]
@abhishekpatel946 i ran the project and it didnt produce apikey missing error. the warning above which says 'it looks like you're using...' was unrelated to the apikey error. it comes for instance, when you import  'firebase performance' but you arent using it in your project anywhere.

if the project produces this missing apikey error when a user is loggedin ; i couldnt reproduce it as there was no signup option in the project. in case i need to be logged in, give me a valid user and password that u used for testing it.

**Comment by abhishekpatel946** [comment]
@atom19-i The userId and password available in the ReadMe for testing. And 'firebase performance' used for monitoring by GCP no worries.
If any kind of error you see in the console while testing the app on your site then let me know with error message along with the screenshot. It's pleasure to be helpful.

**Comment by Yannickdjoa** [comment]
hello @abhishekpatel946  how did you solve this issue? I'm running into the same and I'm not sure what is the solution
